packages/admin/admin/src/common/buttons/save/SaveButton.tsx

```diff
@@ -1,14 +1,65 @@
 import { Check, Error, Error as ErrorIcon, Save, ThreeDotSaving } from "@comet/admin-icons";
-import { Button, ButtonClassKey, ButtonProps, ComponentsOverrides, Theme } from "@mui/material";
-import { WithStyles, withStyles } from "@mui/styles";
-import { ClassKeyOfStyles } from "@mui/styles/withStyles";
-import { ClassNameMap } from "@mui/styles/withStyles/withStyles";
+import { Button, ButtonClassKey, buttonGroupClasses, ButtonProps, ComponentsOverrides } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
 import * as React from "react";
 import { FormattedMessage } from "react-intl";
 
 import { messages } from "../../../messages";
 import { useSplitButtonContext } from "../split/useSplitButtonContext";
-import { SaveButtonClassKey, styles } from "./SaveButton.styles";
+
+export type SaveButtonClassKey = "root" | "saving" | "error" | "success" | "conflict" | ButtonClassKey;
+
+type OwnerState = { savingState?: boolean; errorsState?: boolean; successState?: boolean; conflictState?: boolean };
+
+const Root = styled(Button, {
+    name: "CometAdminSaveButton",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [
+            styles.root,
+            ownerState.savingState && styles.saving,
+            ownerState.errorsState && styles.error,
+            ownerState.successState && styles.success,
+            ownerState.conflictState && styles.conflict,
+        ];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ ownerState, theme }) => css`
+        ${ownerState.savingState &&
+        css`
+            &:disabled {
+                color: ${theme.palette.primary.contrastText};
+                background-color: ${theme.palette.primary.main};
+            }
+            // TODO "&$disabled$containedPrimary" and "&$disabled$containedSecondary"
```

Comment:
For the styles that were previously in `$containedPrimary`, they should just be added inside the `:disabled` if the `variant` is "contained" and the `color` is "primary". You can add the `variant` and `color` props to the `ownerState` so you can access them here. 

```ts
&:disabled {
    // ...

    ${ownerState.variant === "contained" && ownerState.color === "primary" && css`
        // Styles from "&$disabled$containedPrimary"
    `}
}
```


https://mui.com/material-ui/api/button/#classes

<img width="759" alt="image" src="https://github.com/vivid-planet/comet/assets/6264317/ab978aa5-1130-42fe-8600-938205e64fa6">

Reply:
thank you for the explanation!